### PR TITLE
Fixes Signup form

### PR DESCRIPTION
**Pull request by dmartzol**

### CODE REVIEW

File: frontend/src/components/Signup/Signup.jsx [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
+      // If form is valid, continue submission.
+      console.log("Form is legit");
+    } else {
+      // If form is not valid, check which fields are incorrect:
+      formSchema.validate(data, { abortEarly: false }).catch((err) => {
+        console.log(err);
+      });
+    }
 
+    alert(JSON.stringify(data));
```
Comment (weehongayden) [inline frontend/src/components/Signup/Signup.jsx:34]: Did you forget to remove this `alert`?

File: frontend/src/components/Signup/Signup.jsx [hunk 2]
```diff
@@ -41,13 +66,21 @@ function Signup() {
...
                       name="email"
-                      id="email"
                       autoComplete="email"
                       className="block flex-1 border-0 bg-transparent py-1.5 pl-1 text-gray-900 placeholder:text-gray-400 focus:ring-0 sm:text-sm sm:leading-6"
                       placeholder="andrew@example.com"
+                      {...register("email", {
+                        required: true,
+                        pattern:
+                          /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+                      })}
```
Comment (weehongayden) [inline frontend/src/components/Signup/Signup.jsx:79]: Why do you need to perform a validation check here when Yup is already handling it for you?
```js
email: Yup.string()
      .required('Email is required')
      .matches(/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/, 'Invalid email format')
```

File: frontend/src/components/Signup/Signup.jsx (hunk 1 above)
Comment (dmartzol) [reply frontend/src/components/Signup/Signup.jsx:34]: I didn't, but I will have to remove it before the PR is ready. Right now I'm trying to display errors in specific fields when the validation for that field fails

File: frontend/src/components/Signup/Signup.jsx (hunk 2 above)
Comment (dmartzol) [reply frontend/src/components/Signup/Signup.jsx:79]: Thank you, I didnt know how to do this

**Pull request merged**